packages/transactions/src/serializers/message.ts

```diff
@@ -100,12 +91,20 @@ export function getCompiledMessageEncoder(): VariableSizeEncoder<CompiledMessage
 }
 
 export function getCompiledMessageDecoder(): VariableSizeDecoder<CompiledMessage> {
-    return mapDecoder(getStructDecoder(getPreludeStructDecoderTuple()), ({ addressTableLookups, ...restOfMessage }) => {
-        if (restOfMessage.version === 'legacy' || !addressTableLookups?.length) {
-            return restOfMessage;
-        }
-        return { ...restOfMessage, addressTableLookups } as Exclude<CompiledMessage, { readonly version: 'legacy' }>;
-    });
+    return mapDecoder(
+        getStructDecoder(getPreludeStructDecoderTuple()) as VariableSizeDecoder<
+            CompiledMessage & { addressTableLookups?: ReturnType<typeof getCompiledAddressTableLookups> }
```

Comment:
The `ReturnType<..>` could be directly replaced with an `AddressTableLookup[]` here.

Reply:
I'm guessing it was designed that way because the return type is not exported.